src/Umbraco.Web.BackOffice/Security/BackOfficeExternalLoginProviders.cs

```diff
@@ -66,4 +98,26 @@ public bool HasDenyLocalLogin()
         var found = _externalLogins.Values.Where(x => x.Options.DenyLocalLogin).ToList();
         return found.Count > 0;
     }
+
+    /// <inheritdoc />
+    public void InvalidateSessionsIfExternalLoginProvidersChanged()
+    {
+        var previousExternalLoginProvidersValue = _keyValueService.GetValue(ExternalLoginProvidersKey);
+        var currentExternalLoginProvidersValue = string.Join("|", _externalLogins.Keys);
```

Comment:
The order matters here - `_externalLogins` is not an ordered dictionary. It's likely enough to order the keys explicitly here, something like `_externalLogins.Keys.OrderBy(key => key)`